Zend/tests/readonly_props/gh7942.phpt

```diff
@@ -0,0 +1,22 @@
+--TEST--
+GH-7942: Disallow assigning reference to unset readonly property
+--FILE--
+<?php
+
+class Foo {
+    public readonly int $bar;
+    public function __construct(int &$bar) {
+        $this->bar = &$bar;
+    }
+}
+
+try {
+    $i = 42;
+    new Foo($i);
+} catch (Error $e) {
+    echo $e->getMessage(), PHP_EOL;
+}
+
+?>
+--EXPECT--
+Typed property Foo::$bar must not be accessed before initialization
```

Comment:
the error message does not hint that the cause of the problem is related to the by-reference parts.. it feels misleading as is?

Reply:
Assuming this results in a BP_VAR_W read_property call, we should be able to improve the error message based on that and say something about indirect modification of readonly property being forbidden, or so.